src/main/java/org/radarcns/management/service/mapper/SourceMapper.java

```diff
@@ -39,11 +39,13 @@
 
     @Mapping(source = "sourceId" , target = "sourceId")
     @Mapping(source = "expectedSourceName" , target = "expectedSourceName")
+    @Mapping(source = "deviceType.id" , target = "deviceTypeId")
     @Mapping(source = "deviceType.deviceModel" , target = "deviceTypeModel")
     @Mapping(source = "deviceType.deviceProducer" , target = "deviceTypeProducer")
     @Mapping(source = "deviceType.catalogVersion" , target = "deviceCatalogVersion")
     @Mapping(source = "project.id" , target = "projectId")
     @Mapping(source = "assigned" , target = "assigned")
```

Comment:
Mapping of same attributes will work with or without Mapping annotation.

Reply:
Thanks, I wasn't aware